packages/react-core/src/components/Banner/Banner.tsx

```diff
@@ -1,39 +1,68 @@
 import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Banner/banner';
 import { css } from '@patternfly/react-styles';
+import { variantIcons } from '../Alert/AlertIcon';
+import { Flex, FlexItem } from '../../layouts';
 
 export interface BannerProps extends React.HTMLProps<HTMLDivElement> {
-  /** Content rendered inside the banner */
+  /** Content rendered inside the banner. */
   children?: React.ReactNode;
-  /** Additional classes added to the banner */
+  /** Additional classes added to the banner. */
   className?: string;
-  /** Variant styles for the banner */
-  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
+  /** A custom icon for the banner. This property will override the icon that is set based on
+   * the variant property.
+   */
+  customIcon?: React.ReactNode;
+  /** Flag for indicating whether the banner has a status icon. When set to "true", the icon
+   * will be set based on the variant property.
+   */
+  hasStatusIcon?: boolean;
   /** If set to true, the banner sticks to the top of its container */
   isSticky?: boolean;
-  /** Text announced by screen readers to indicate the type of banner. Defaults to "${variant} banner" if this prop is not passed in */
+  /** Text announced by screen readers to indicate the type of banner when the hasStatusIcon property
+   * is passed in. Defaults to "${variant} banner" if this property is not passed in.
+   */
   screenReaderText?: string;
+  /** Variant styles for the banner. */
+  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
 }
 
 export const Banner: React.FunctionComponent<BannerProps> = ({
   children,
   className,
+  customIcon,
+  hasStatusIcon = false,
   variant = 'default',
   screenReaderText,
   isSticky = false,
   ...props
-}: BannerProps) => (
-  <div
-    className={css(
-      styles.banner,
-      styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-      isSticky && styles.modifiers.sticky,
-      className
-    )}
-    {...props}
-  >
-    {children}
-    <span className="pf-u-screen-reader">{screenReaderText || `${variant} banner`}</span>
-  </div>
-);
+}: BannerProps) => {
+  const StatusIcon = variantIcons[variant];
+
+  return (
+    <div
+      className={css(
+        styles.banner,
+        styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
+        isSticky && styles.modifiers.sticky,
+        className
+      )}
+      {...props}
+    >
+      {hasStatusIcon ? (
+        <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+          <FlexItem>
+            <span className="pf-u-screen-reader">{screenReaderText || `${variant} banner`}</span>
+            {customIcon || <StatusIcon />}
+          </FlexItem>
+          <FlexItem>
+            <div className="pf-l-flex__item">{children}</div>
+          </FlexItem>
+        </Flex>
+      ) : (
```

Comment:
I should have made a note about this, but we currently don't manage the layout of the banner contents at all. All of our status examples should really be demos, since we're adding a flex layout to space the text and icon. In practice,  it's up to the user to manage the layout and spacing and everything. For example, they may want to have a banner like this:

```
-----------------------------------------------------------
| Some text          [icon] Alert text          More text |
-----------------------------------------------------------
```

You can see some examples in the design docs - https://www.patternfly.org/v4/components/banner/design-guidelines#when-to-use

So they'll need to be able to choose the layout and spacing and everything. I think if we have some examples, that should be good. cc @mcarrano does that all sound right?

Reply:
That's a good point. In that case, we may not even want to have a prop that determines whether an icon gets rendered, but rather just show an example of banners to convey status? In that case the example would just have the icon and banner text passed in as children of Banner.

Reply:
@thatblindgeye yep!

Reply:
So, should this not be in the demo section rather than an example.